From d50c94f1adba49c760ca4dddc139a26855206ed9 Mon Sep 17 00:00:00 2001
From: Johannes Dold <17587766+johannesdold@users.noreply.github.com>
Date: Wed, 2 Jun 2021 09:15:57 +0200
Subject: [PATCH] 8300 Increase query column and column format to varchar 64,
 add exception if provided quantity value id exceeds the restriction from the
 database (#9250)

---
 .../Controller/Admin/DataObject/QuantityValueController.php   | 4 ++++
 models/DataObject/ClassDefinition/Data/QuantityValue.php      | 4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/bundles/AdminBundle/Controller/Admin/DataObject/QuantityValueController.php b/bundles/AdminBundle/Controller/Admin/DataObject/QuantityValueController.php
index 8653523a8e..b62cbcabe3 100644
--- a/bundles/AdminBundle/Controller/Admin/DataObject/QuantityValueController.php
+++ b/bundles/AdminBundle/Controller/Admin/DataObject/QuantityValueController.php
@@ -134,6 +134,10 @@ class QuantityValueController extends AdminController
                     throw new \Exception('unit with ID [' . $id . '] already exists');
                 }
 
+                if (mb_strlen($id) > 50) {
+                    throw new \Exception('The maximal character length for the unit ID is 50 characters, the provided ID has ' . mb_strlen($id) . ' characters.');
+                }
+
                 $unit = new Unit();
                 $unit->setValues($data);
                 $unit->save();
diff --git a/models/DataObject/ClassDefinition/Data/QuantityValue.php b/models/DataObject/ClassDefinition/Data/QuantityValue.php
index 924d564be3..75a5e794a4 100644
--- a/models/DataObject/ClassDefinition/Data/QuantityValue.php
+++ b/models/DataObject/ClassDefinition/Data/QuantityValue.php
@@ -98,7 +98,7 @@ class QuantityValue extends Data implements ResourcePersistenceAwareInterface, Q
      */
     public $queryColumnType = [
         'value' => 'double',
-        'unit' => 'varchar(50)',
+        'unit' => 'varchar(64)',
     ];
 
     /**
@@ -110,7 +110,7 @@ class QuantityValue extends Data implements ResourcePersistenceAwareInterface, Q
      */
     public $columnType = [
         'value' => 'double',
-        'unit' => 'varchar(50)',
+        'unit' => 'varchar(64)',
     ];
 
     /**
-- 
2.30.1 (Apple Git-130)

